python/src/deltachat/cutil.py

```diff
@@ -16,4 +16,9 @@ def iter_array(dc_array_t, constructor):
 
 
 def from_dc_charpointer(obj):
-    return ffi.string(obj).decode("utf8")
+    # already creates a copy in python land
+    s = ffi.string(obj).decode("utf8")
+    # free the cstring
+    lib.dc_str_unref(obj)
```

Comment:
```suggestion
    return ffi.gc(obj, lib.dc_str_unref).decode("utf8") 
```
is the typical pattern for making the ffi-glue code care for calling the proper unref function when the ``obj`` goes out of scope, i.e. is gc-collected by python.  i wasn't aware that these dc_charpointer's need a special unref function.

Reply:
um, wait a second. that's not a working suggestion, was a bit too fast.  you still need to call ffi.string on the ffi.gc'ed object.  or you could do the lib.dc_str_unref in a "finally: " clause so that it's also executed if the decode-utf8 fails.